### DISCUSSION TITLE
Design a pen 🖊️

### DISCUSSION THREAD

**Discussion by kanmaytacker** [kanmaytacker/fundamentals#23] [General]
**What?**
Your task is to design a set of classes to simulate a pen. Your class design should have common attributes and functionalities of a pen and follow good design principles

**Why?**
Practising software design.

**How?**

**Requirements**
- A pen is anything that can write.
- Pen can be Gel, Ball, Fountain, Marker.
- Ball Pen and Gel Pen have a Ball Pen Refill and a Gel Pen Refill respectively to write.
- A refil has a tip and an ink.
- Ink can be of different colour
- A fountain pen has an Ink.
- Refil has a radius.
- For fountain pen, its tip has a radius.
- Each pen can write in a different way.
- Some pens write in the same way.
- Every pen has a brand and a name.
- Some pens may allow refilling while others might not.

**Approach**
- Identify the major classes you would need to create
- For each class, list the attributes and methods needed.
- Create a class diagram

**Submission**
- The objective is to think about the class design. Don't focus on the submission format
- You can literally just post a list of all classes, their attributes and methods.
- The ideal solution is a class diagram which can be drawn on paper or using tools like [draw.io](https://app.diagrams.net/).

**Comment by MakaBaca** [comment]
Pen Class Diagram

![Pen_Class_Diagram](https://github.com/kanmaytacker/fundamentals/assets/42783016/69028842-1ab8-494c-b9c7-5657fc005ea8)
